src/QMCHamiltonians/tests/test_paircorr.cpp

```diff
@@ -0,0 +1,197 @@
+                        </simulationcell>";
+
+// Particleset block
+const char *pset_xml = "<particleset name=\"e\" random=\"yes\"> \
+                          <group name=\"u\" size=\"4\" mass=\"1.0\"> \
+                            <parameter name=\"charge\" >   -1  </parameter> \
+                            <parameter name=\"mass\"   >  1.0  </parameter> \
+                          </group> \
+                          <group name=\"d\" size=\"4\" mass=\"1.0\"> \
+                            <parameter name=\"charge\" >   -1  </parameter> \
+                            <parameter name=\"mass\"   >  1.0  </parameter> \
+                          </group> \
+                        </particleset>";
+
+// PairCorrEstimator block
+const char* gofr_xml = "<estimator type=\"gofr\" name=\"gofr\" rmax=\"2.0\" num_bin=\"20\" />";
+
+
+namespace qmcplusplus
+{
+TEST_CASE("Pair Correlation", "[hamiltonian]")
+{
+  std::cout << std::fixed;
+  std::cout << std::setprecision(8);
+  typedef QMCTraits::RealType RealType;
+
+  Communicate* c;
+  OHMMS::Controller->initialize(0, NULL);
+  c = OHMMS::Controller;
+
+  CrystalLattice<OHMMS_PRECISION, OHMMS_DIM> Lattice;
+  Lattice.BoxBConds = true; // periodic
+  Lattice.R.diagonal(2.0);
+  Lattice.reset();
+
+  // XML parser
+  Libxml2Document doc;
+
+  std::cout << "\n\n\ntest_paircorr:: START\n";
+
+  // TEST new idea: ParticlesetPool to make a ParticleSet
+  bool lat_okay = doc.parseFromString(lat_xml);
+  REQUIRE(lat_okay);
+  xmlNodePtr lat_xml_root = doc.getRoot();
+
+  ParticleSetPool pset_builder(c, "pset_builder");
+  pset_builder.putLattice(lat_xml_root); // Builds lattice
+
+  bool pset_okay = doc.parseFromString(pset_xml);
+  REQUIRE(pset_okay);
+  xmlNodePtr pset_xml_root = doc.getRoot();
+  pset_builder.put(pset_xml_root);  // Builds ParticleSet
+
+  // Get the (now assembled) ParticleSet, do simple sanity checks, then print info 
+  ParticleSet* elec = pset_builder.getParticleSet("e");
+  elec->Lattice.copy(Lattice); // copy in the new Lattice
+
+  REQUIRE(elec->SameMass);
+  REQUIRE(elec->getName() == "e");
+
+  // Move the particles manually onto B1 lattice
+  // NB: Spins are grouped contiguously
+  // Up spins
+  elec->R[0][0] = 0.0;
+  elec->R[0][1] = 0.0;
+  elec->R[0][2] = 0.0;
+  elec->R[1][0] = 1.0;
+  elec->R[1][1] = 1.0;
+  elec->R[1][2] = 0.0;
+  elec->R[2][0] = 1.0;
+  elec->R[2][1] = 0.0;
+  elec->R[2][2] = 1.0;
+  elec->R[3][0] = 0.0;
+  elec->R[3][1] = 1.0;
+  elec->R[3][2] = 1.0;
+
+  // Down spins
+  elec->R[4][0] = 1.0;
+  elec->R[4][1] = 0.0;
+  elec->R[4][2] = 0.0;
+  elec->R[5][0] = 0.0;
+  elec->R[5][1] = 1.0;
+  elec->R[5][2] = 0.0;
+  elec->R[6][0] = 0.0;
+  elec->R[6][1] = 0.0;
+  elec->R[6][2] = 1.0;
+  elec->R[7][0] = 1.0;
+  elec->R[7][1] = 1.0;
+  elec->R[7][2] = 1.0;
+
+  elec->get(std::cout); // print particleset info to stdout
+
+  // Set up the distance table
+  const int ee_table_id = elec->addTable(*elec, DT_AOS, false);
```

Comment:
This should have an ifdef for SOA vs. AOS
```
#ifdef ENABLE_SOA
 ....  addTable(*eloc, DT_SOA, false);
#else
  ...  addTable(*eloc, DT_AOS, false);
#endif

Reply:
Wow. Turns out if you run in SoA mode, you can still create a AoS distance table. For the unit test, this meant that PairCorrEstimator basically never went down the SoA path in PairCorrEstimator::evaluate() !

Fixed and pushing new commit now.